hocs/withTracking.js

```diff
@@ -0,0 +1,20 @@
+import useTrackingEvents from "hooks/analytics/useTrackingEvents";
+
+const withTracking = () => (PageComponent) => {
+  const WithInjectedStores = (props) => {
+    const callTracking = useTrackingEvents();
+    return (
+      <PageComponent {...props} callTracking={callTracking} />
+    );
+  };
+
+    // Set the correct displayName in development
+  if (process.env.NODE_ENV !== "production") {
+    const displayName = PageComponent.displayName || PageComponent.name || "Component";
+    WithInjectedStores.displayName = `withInjectedStores(${displayName})`;
+  }
+
+  return WithInjectedStores;
+};
+
+export default withTracking;
```

Comment:
I guess this was added as a workaround for now to avoid converting `ProductDetail` to a functional component?
`WithInjectedStores` sounds a bit misleading, as essentially, none of the global stores/context is added here.